d3a_interface/matching_algorithms/requirements_validators.py

```diff
@@ -0,0 +1,205 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+
+from abc import ABC, abstractmethod
+from typing import Dict, Optional, Union
+
+from d3a_interface.data_classes import Offer, Bid
+
+
+class Requirement(ABC):
```

Comment:
Nit: since "requirement" is a term that can be used in many different contexts, what do you think of renaming it? My first suggestion would be `OrderRequirement`, since this would refer to both bids and offers as discussed on Slack.